src/synthesized_insight/metrics/base.py

```diff
@@ -83,6 +83,20 @@ def __call__(self, sr_a: pd.Series, sr_b: pd.Series) -> Tuple[Union[int, float,
         return self._compute_test(sr_a, sr_b)
 
 
+class DataFrameMatrix(_Metric):
```

Comment:
Do we need an abstract class here? We should only define one when we need the abstraction, I'm not sure we do here

Reply:
I think having an abstract class is useful for code clarity in this case because the ColumnMap, CorrMatrix, and DiffCorrMatrix have base class as _Metric, but the `__call__` method for each of these returns a dataframe or a Tuple of the Dataframe, whereas every other `_Metric` base class returns either single metric_value or a tuple of metric value and p_val; return type of the `_call_` for these ColumnMap, CorrMatrix, and DiffCorrMatrix classes are different from the rest of the metric classes